Development environment setup

andres-arana opened the pull request:
See https://github.com/SkyTruth/pelagos-server/issues/621.

A couple of changes on the development environment setup:
- Adds a common vagrantfile to the project. This allows us to have all system dependencies (java, xvfb, chromium, etc.) in a virtualized environment, which is configured automatically when you do vagrant up. No environment setup instructions required, you just run vagrant up and everything is installed and configured properly. In addition to this, we prevent "oh, you don't have X in your machine, and you actually require version z.z.z of Y as well" errors.

The idea with this is that we can use a common environment independent on what we have on our machines. Anybody who wants to run the project just needs to clone the repo, vagrant up and they are ready.
- Adds a `Vagrantprov` provisioning script which installs everything we need on the virtual machine.
- Adds the testing commands to the makefile, so that we don't need to document how to run unit tests and update the documentation when anything changes. I also changed the travis scripts so that we use the same commands on CI and on dev.
- Adds README documentation for all this, including links on how to install vagrant, how to run the different, common make tasks we provide and such.

Commit:
Implements a virtualized development environment with vagrant

Adds a vagrantfile to have a virtualized, common development
environment with required dependencies.

Commit:
Updates the development environment to be self-contained

* Adds some new phone makefile tasks to run unit and integration tests,
  and to start a simple development server locally.

* Configures the virtualized environment to have all required
  dependencies and matching how we run it on travis.

* Adds documentation on how to setup, run and test the application.

redhog commented on Makefile:
```diff
@@ -125,3 +125,12 @@ clean-dependencies:
+	xvfb-run -s "-screen 0 1280x1024x24" nosetests -s -w ui_tests
+
+server:
+	python -m SimpleHTTPServer 8080
```
This should run ui_tests/server.py instead, as it also provides an interactive selenium driver environment, and produces test data to play with.

This is useful for developing new integration tests

redhog commented on Vagrantprov:
```diff
@@ -0,0 +1,30 @@
+ #! /bin/bash
+
+# Ensure apt is up to date
```
There is quite some overlap between this and .travis.yml's install section. Maybe we could move this to the makefile and just call it from both places?

Commit:
Refactors the integration test scripts

* Creates a new script to start a development server, previously
  generating test tilesets.

* Refactors the integration test infrastructure to extract the tileset
  generation and server startup out of the integration test themselves.
  This allows us to start a server and generate the tilesets for
  development purposes.

Commit:
Merges the install section in travis with vagrant provision

* This merges the install section in travis.yml with the vagrant
  provisioning script so that environment prerequisites are the same in
  travis as in the local development environment.

Commit:
Merges the development and testing server into one

Commit:
Moves the project dependencies into the makefile

Commit:
Adds documentation on how to run the make tasks
